src/main/java/com/kelinci/mainapp/api/SimpleUserDatabase.java

```diff
@@ -0,0 +1,41 @@
+package com.kelinci.mainapp.api;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+public class SimpleUserDatabase {
+
+    private static final  List<OurUser> userDatabase = new ArrayList<>();
+
+    public static List<OurUser> getUserDatabase() {
+        return userDatabase;
+    }
+
+    public static void addUserToDatabase(OurUser ourUser) {
+        userDatabase.add(ourUser);
+    }
+
+    public void clearUserDatabase() {
+        userDatabase.clear();
+    }
+    public Optional<OurUser> getUserWithMail(String mail){
+        return getUserDatabase().stream()
+                .filter(ourUser -> ourUser.getMail().equals(mail))
+                .findFirst();
+    }
+    public static String generateMailCode() {
```

Comment:
to trzeba będzie przerzucić do jakiejś klasy Utilowej (pomocniczej). Bo czemu niby baza danych ma umieć (znać) jak generować mail code?

Reply:
ok, zrobiona klasa FirstUtilClass. Do niej trafia generateMailCode() - wtedy jest generowany obiekt z tej klasy który daje tą metodę w addRegisteredUser, a także w metodzie z confirm() w main. Czyli sobie generujemy instancje na chwilkę tylko dla tej metody. Już bez staticów znowu.